app/bundles/EmailBundle/Controller/Api/EmailApiController.php

```diff
@@ -131,6 +131,10 @@ public function sendLeadAction($id, $leadId)
             }
 
             $leadFields = array_merge(['id' => $leadId], $lead->getProfileFields());
+            // Set owner_id to support the "Owner is mailer" feature
+            if ($lead->getOwner()) {
+                $leadFields['owner_id'] = $lead->getOwner()->getId();
```

Comment:
When looking at 

https://github.com/mautic/mautic/blob/85eb220c346d11944715172174237f7eca932cd9/app/bundles/EmailBundle/Helper/MailHelper.php#L346

and 

https://github.com/mautic/mautic/blob/85eb220c346d11944715172174237f7eca932cd9/app/bundles/EmailBundle/Helper/MailHelper.php#L2099

can you also set the signature, first and last name and email so the from and signature parts of code would work too?

Reply:
I need to test whether the signature is set properly, because the function `getContactOwnerSignature` https://github.com/mautic/mautic/blob/85eb220c346d11944715172174237f7eca932cd9/app/bundles/EmailBundle/Helper/MailHelper.php#L2094-L2101
is being executed after `setFrom`
https://github.com/mautic/mautic/blob/85eb220c346d11944715172174237f7eca932cd9/app/bundles/EmailBundle/Helper/MailHelper.php#L346-L347

Reply:
Thank you for mentioning the owner signature since I hadn't included it in the test cases. I have pushed a commit to test that the owner signature is used as expected when the owner is the sender. The test results do confirm that the current implementation is correct.